batchupload.php

```diff
@@ -0,0 +1,306 @@
+    );
+    $fs = get_file_storage();
+    $batchuploadedfiles = $fs->get_area_files($coursecontext->id, 'block_opencast', upload_helper::OC_FILEAREA, $itemid);
+
+    // Cleanup the drafts.
+    $usercontext = \context_user::instance($USER->id);
+    $draftfiles = $fs->get_area_files($usercontext->id, 'user', 'draft', $data->batchuploadedvideos, 'id', false);
+    if (!empty($draftfiles)) {
+        foreach ($draftfiles as $draftfile) {
+            file_deletionmanager::fulldelete_file($draftfile);
+        }
+    }
+
+    // Preparing batch metadata to use.
+    $metadata = [];
+
+    if (property_exists($data, 'series')) {
+        $metadata[] = [
+            'id' => 'isPartOf',
+            'value' => $data->series,
+        ];
+    }
+
+    // Adding data into $metadata based on $metadata_catalog.
+    foreach ($batchmetadatacatalog as $field) {
+        $id = $field->name;
+        if (property_exists($data, $field->name) && $data->$id) {
+            if ($field->name == 'subjects') {
+                !is_array($data->$id) ? $data->$id = [$data->$id] : $data->$id = $data->$id;
+            }
+            $obj = [
+                'id' => $id,
+                'value' => $data->$id,
+            ];
+            $metadata[] = $obj;
+        }
+    }
+
+    $sd = new DateTime("now", new DateTimeZone("UTC"));
+    $sd->setTimestamp($data->startDate);
+    $startdate = [
+        'id' => 'startDate',
+        'value' => $sd->format('Y-m-d'),
+    ];
+    $starttime = [
+        'id' => 'startTime',
+        'value' => $sd->format('H:i:s') . 'Z',
+    ];
+    $metadata[] = $startdate;
+    $metadata[] = $starttime;
+
+    // Prepare the visibility object.
+    $visibility = new stdClass();
+    $visibility->initialvisibilitystatus = !isset($data->initialvisibilitystatus) ?
+        block_opencast_renderer::VISIBLE : $data->initialvisibilitystatus;
+    $visibility->initialvisibilitygroups = !empty($data->initialvisibilitygroups) ?
+        json_encode($data->initialvisibilitygroups) : null;
+    // Check if the scheduled visibility is set.
+    if (isset($data->enableschedulingchangevisibility) && $data->enableschedulingchangevisibility) {
+        $visibility->scheduledvisibilitytime = $data->scheduledvisibilitytime;
+        $visibility->scheduledvisibilitystatus = $data->scheduledvisibilitystatus;
+        $visibility->scheduledvisibilitygroups = !empty($data->scheduledvisibilitygroups) ?
+            json_encode($data->scheduledvisibilitygroups) : null;
+    }
+
+    $error = null;
+    $totalfiles = count($batchuploadedfiles);
+    // Loop through the files and proceed with the upload and cleanup records.
+    if (!empty($batchuploadedfiles)) {
+        $errorcount = 0;
+        foreach ($batchuploadedfiles as $uploadedfile) {
+            try {
+                if ($uploadedfile->get_filename() === '.') {
+                    file_deletionmanager::fulldelete_file($uploadedfile);
+                    $totalfiles--;
```

Comment:
I might have overlooked something, but this isn't needed, right?

Reply:
You mean the whole '.' file name checker?
If yes, it is necessary here!

Reply:
No, sorry, I just meant the `$totalfiles` variable and decrement

Reply:
The `$totalfiles` variable is used for success info and decrement is also important to show the accurate file count result!

Reply:
You're right, I somehow overread that. Thanks!